core/templates/cowdata.h

```diff
@@ -436,20 +436,19 @@ Error CowData<T>::_realloc(Size p_alloc_size) {
 
 template <typename T>
 typename CowData<T>::Size CowData<T>::find(const T &p_val, Size p_from) const {
-	Size ret = -1;
+	const Size s = size();
 
-	if (p_from < 0 || size() == 0) {
-		return ret;
+	if (p_from < 0 || s == 0) {
+		return -1;
 	}
 
-	for (Size i = p_from; i < size(); i++) {
-		if (get(i) == p_val) {
-			ret = i;
-			break;
+	for (Size i = p_from; i < s; i++) {
```

Comment:
Maybe do this ? 

```
if (p_from < 0) {
return -1;
}

const Size s = size();
```
As getting the size is not free, and if it's 0, it will go over the loop anyway. 
Not sure it will make any difference.

Reply:
It should be marginal but will test

Reply:
Didn't make any measurable change as far as I could tell, but will change anyway, should just be within the error margins

Reply:
The == 0 check is unnecessary, the loop performs the check for you. if it's an unnecessary additional branch (that might even be optimized out by the compiler).

Reply:
Good point I'll remove that too